text/0000-edition-2021.md

```diff
@@ -0,0 +1,289 @@
+* **Manual changes:** As the final step, resolve any remaining lints or errors manually (hopefully these will be quite limited). Idiom lints, in particular, may not have automated suggestions, though you can always add an `#![allow]` at the crate level to silence them if you are in a hurry.
+
+## Specifying edition changes and migration plans
+
+Any RFC or other proposal that will cause code that used to compile on edition `N` no longer compiles on edition `N+1` must include detais of how the "current edition" for that change is determined as well a migration plan. 
+
+### Determining the current edition
+
+Altough there is an "ambient" edition for the crate that is specified as part of `Cargo.toml`, individual bits of code can be tied to earlier editions due to the interactions of macros. For example, if a Rust 2021 crate invokes a macro from a Rust 2018 crate, the body of that macro may be interpreted using Rust 2018 rules. For this reason, whenever an edition change is proposed, it's important to specify the tokens from which the edition is determined.
+
+As an example, if we are introducing a new keyword, then the edition will be taken from the keyword itself. If we were to make a change to the semantics of the `+` operator, we might say that the current edition is determined from the `+` token. This way, if a macro author were to write `$a + $b`, then this expression would use the edition rules from the macro definition (which contained the `+` token). Additions that defined in the `$a` expression would use the edition from which `$a` was derived.
+
+### Migration plans
+
+These migration plans should be structured as the answer to three questions; these questions are designed to compartmentalize review and implementation.
+
+The three questions are:
+
+* What code patterns no longer compile (or change meaning) in the new edition?
+* Which code patterns will you migrate from the old to the new edition?
+    * This should be some subset of the answer to the first question.
+* What is your plan to migrate each code pattern?
+    * The plan should include automated suggestions targeting the intersection of Edition N and Edition N+1.
+    * The plan may also include lints specific to Edition N+1 that will cleanup the code to make it more idiomatic.
+
+More details on these questions follow.
+
+#### What code patterns no longer compile (or change meaning) in the new edition?
+
+The RFC should list **all** known code patterns that are affected by the change in edition, either because they no longer compile or because their meaning changes. This should include unusual breakage that is not expected in practice. 
+
+Answering this question is not a commitment to automatically migrate every instance of these code atterns, but it is important for this research to be done during the design phase such that it can be considered when evaluating migration strategies.
```

Comment:
```suggestion
Answering this question is not a commitment to automatically migrate every instance of these code patterns, but it is important for this research to be done during the design phase such that it can be considered when evaluating migration strategies.
```